sdcm/utils/replication_strategy_utils.py

```diff
@@ -1,45 +1,69 @@
 import ast
+import logging
 import re
 
 from contextlib import ContextDecorator
-from typing import Callable, Dict
+from typing import Callable, Dict, TYPE_CHECKING
 
-from sdcm.cluster import BaseNode
 from sdcm.utils.cql_utils import cql_quote_if_needed
+if TYPE_CHECKING:
+    from sdcm.cluster import BaseNode
+
+LOGGER = logging.getLogger(__name__)
 
 
 class ReplicationStrategy:  # pylint: disable=too-few-public-methods
 
     @classmethod
     def from_string(cls, replication_string):
-        replication_value = re.search(r".*replication[\s]*=[\s]*(\{.*\})", replication_string, flags=re.IGNORECASE)
+        # To solve the problem when another curly braces were added (tablets related).
+        # Example:
+        # CREATE KEYSPACE scylla_bench WITH replication = {'class': 'org.apache.cassandra.locator.SimpleStrategy',
+        # 'replication_factor': '1'} AND durable_writes = true AND tablets = {'enabled': false};
+        LOGGER.debug("Analyze replication string '%s'", replication_string)
```

Comment:
this one might work better `.*replication[\s]*=[\s]*(\{.*?\})`

Reply:
changed